code.js

```diff
@@ -1,3 +1,76 @@
-function augmentingPath(graph, start, end) {
-    return [];
+function augmentingPath(graph, start, end){
+	fullPath=[]
+	visitedNodes=[]
+	keyValues=Object.keys(graph);
+	source=keyValues[0]
+	for (var i=0; i<Object.keys(graph).length; i++){
+		visitedNodes[keyValues[i]]= Infinity
+	}
+	visitedNodes[source]=0
+	returnedValue=dAlgorithm(graph, start, end);
+	if (returnedValue==0)
+		return fullPath
+	else if (returnedValue==1)
+		return []
+	else if (returnedValue==2)
+		return []
+	console.log(visitedNodes)
+	//console.log(fullPath)
+	//return fullPath
 }
+
+function dAlgorithm(graph, currentNode, goalNode){
```

Comment:
Not sure what's going on here, but this looks way more complex than what you need. You don't need to consider the weights or anything like that. It also looks like you're keeping track of the visited nodes here again?